File: src/main/java/com/leesh/inflpick/common/adapter/in/web/ErrorCode.java
package com.leesh.inflpick.common.adapter.in.web;

import lombok.Getter;

public enum ErrorCode {

    CO_0001("유저 입력값이 올바르지 않음"),
    ;

    @Getter
    private final String cause;

    ErrorCode(String cause) {
        this.cause = cause;
    }
}

File: src/main/kotlin/com/leesh/inflpick/common/exception/ErrorCode.kt
package com.leesh.inflpick.common.exception

import org.springframework.http.HttpStatus

enum class ErrorCode(val status: Int, val code: String, val reason: String, val comment: String, val action: String) {

    MISSING_REQUIRED_PARAMETER(HttpStatus.BAD_REQUEST.value(), "MISSING_REQUIRED_PARAMETER", "API 요청 필수 파라미터가 입력되지 않았습니다.", "API 요청 필수 파라미터가 입력되지 않은 경우에 발생", "API 요청 필수 파라미터 입력 후 재시도"),
    ;

    fun toErrorMessage(): String {
        return """
                  ErrorCode = [$code]
                  Reason = $reason
                  Comment = $comment
                  Action = $action
               """
    }
}